map-reduce/word-count.c

```diff
@@ -127,45 +150,60 @@ static inline struct ht_node *ht_find(struct htable *h, void *key)
 /* Insert a new element with the key 'key' in the htable.
  * Return 0 if success.
  */
+#include <stdio.h>
+
 static inline int ht_insert(struct htable *h, struct ht_node *n, void *key)
 {
     hash_t hval;
     uint32_t bkt;
     h->hashfunc(key, &hval, &bkt);
     n->hash = hval;
 
-    struct list_entry *head = &h->buckets[bkt], *iter;
-    list_for_each (head, iter) {
-        struct ht_node *tmp = list_element(iter, struct ht_node, list);
+    struct hlist_head *head = &h->buckets[bkt];
+    struct hlist_node *iter;
+    hlist_for_each(iter, head)
```

Comment:
Update `.clang-format` to recognize `hlist_for_each` as a for-each macro.

Reply:
Done!

Reply:
You shall indent this source file accordingly.